diff --git a/main.js b/main.js
index d10535a04e2f609fe45cb8f563952bccaa6b04a8..1f29ad2aba16edc84d87246db77d79581032e802 100755
--- a/main.js
+++ b/main.js
@@ -32031,6 +32031,10 @@ var _PackageJson = class _PackageJson extends Class3("PackageJson")({
     provenance: optionalWith(Boolean$, { default: () => false }),
     executableFiles: optional(Array$(String$))
   })),
+  homepage: optional(String$),
+  keywords: optional(Array$(String$)),
+  bugs: optional(Record({ key: String$, value: String$ })),
+  engines: optional(Record({ key: String$, value: String$ })),
   license: String$,
   author: optional(
     Union2(
@@ -32311,6 +32315,10 @@ var run11 = gen2(function* () {
     addOptional("gitHead");
     addOptional("bin");
     addOptional("bin");
+    addOptional("keywords");
+    addOptional("homepage");
+    addOptional("bugs");
+    addOptional("engines");
     if (ctx.packageJson.publishConfig?.provenance === true) {
       out.publishConfig = { provenance: true };
     }
