diff --git a/main.js b/main.js
index d10535a04e2f609fe45cb8f563952bccaa6b04a8..b83374993825e50a9ae3068c4da8605cf240de81 100755
--- a/main.js
+++ b/main.js
@@ -32031,6 +32031,10 @@ var _PackageJson = class _PackageJson extends Class3("PackageJson")({
     provenance: optionalWith(Boolean$, { default: () => false }),
     executableFiles: optional(Array$(String$))
   })),
+  homepage: optional(String$),
+  keywords: optional(Array$(String$)),
+  bugs: optional(Record({ key: String$, value: String$ })),
+  engines: optional(Record({ key: String$, value: String$ })),
   license: String$,
   author: optional(
     Union2(
@@ -32459,6 +32463,10 @@ var run12 = gen2(function* () {
     addOptional("optionalDependencies");
     addOptional("gitHead");
     addOptional("bin");
+   addOptional("keywords");
+   addOptional("homepage");
+   addOptional("bugs");
+   addOptional("engines");
     if (ctx.packageJson.publishConfig?.provenance === true) {
       out.publishConfig = { provenance: true };
     }
