diff --git a/bin.cjs b/bin.cjs
index edbbe365217db0ed042cb6189a533dc41c7114ce..e4abc5809fdc872d8f135f77f437f7ce5686b103 100755
--- a/bin.cjs
+++ b/bin.cjs
@@ -2199,7 +2199,7 @@ var require_utils = __commonJS({
         return options4.slugify(str, options4);
       }
       str = utils.getTitle(str);
-      str = utils.stripColor(str);
+      // str = utils.stripColor(str);
       str = str.toLowerCase();
       str = str.split(" ").join("-");
       str = str.split(/\t/).join("--");
