diff --git a/dist/esm/internal/http/nodeClient.js b/dist/esm/internal/http/nodeClient.js
index 194d8f007098661c64ff85387c1190a5f5b1e70f..35fa04c4650e46c0ed88b3ea6213a97223c6094e 100644
--- a/dist/esm/internal/http/nodeClient.js
+++ b/dist/esm/internal/http/nodeClient.js
@@ -103,9 +103,11 @@ const waitForResponse = (nodeRequest, request) => Effect.async(resume => {
     nodeRequest.off("error", onError);
     resume(Effect.succeed(response));
   }
+  nodeRequest.on("upgrade", onResponse);
   nodeRequest.on("response", onResponse);
   return Effect.sync(() => {
     nodeRequest.off("error", onError);
+    nodeRequest.off("upgrade", onResponse);
     nodeRequest.off("response", onResponse);
   });
 });
