diff --git a/main.js b/main.js
index f478f2e32337e478c7cf92c78ad4e1e2a2881f11..4fac849420a499fae147ff97b658cd625c1945e1 100755
--- a/main.js
+++ b/main.js
@@ -29895,9 +29895,18 @@ var _PackageJson = class _PackageJson extends Class2()({
   version: string5,
   description: string5,
   private: optional(boolean2, { default: () => false }),
-  publishConfig: optional(struct6({
-    provenance: optional(boolean2, { default: () => false })
-  })),
+  publishConfig: optional(
+    struct6({
+      provenance: optional(boolean2, { default: () => false }),
+      executableFiles: optional(array6(string5), {
+        default: () => [],
+      }),
+    })
+  ),
+  homepage: optional(string5),
+  keywords: optional(array6(string5)),
+  bugs: optional(record2(string5, string5)),
+  engines: optional(record2(string5, string5)),
   license: string5,
   author: optional(string5),
   repository: union12(
@@ -30130,9 +30139,22 @@ var run11 = gen2(function* (_) {
     addOptional("gitHead");
     addOptional("bin");
     addOptional("bin");
+    addOptional("keywords");
+    addOptional("homepage");
+    addOptional("bugs");
+    addOptional("engines");
     if (ctx.packageJson.publishConfig?.provenance === true) {
       out.publishConfig = { provenance: true };
     }
+    if (
+      ctx.packageJson.publishConfig?.executableFiles !== undefined
+      && ctx.packageJson.publishConfig.executableFiles.length > 0
+    ) {
+      out.publishConfig = {
+        ...out.publishConfig,
+        executableFiles: ctx.packageJson.publishConfig.executableFiles,
+      }
+    }
     if (ctx.hasMainCjs) {
       out.main = "./dist/cjs/index.js";
     }
