diff --git a/lib/utilities/Constants.js b/lib/utilities/Constants.js
index a98afe7eddb0c388d9666d80c715c0fec91bae39..756584fd9c92d84fdd8aa2844e71b93406abec65 100644
--- a/lib/utilities/Constants.js
+++ b/lib/utilities/Constants.js
@@ -6,7 +6,7 @@ exports.Constants = void 0;
 class Constants {
 }
 exports.Constants = Constants;
-Constants.projectConfigFolderName = 'config';
+Constants.projectConfigFolderName = './';
 Constants.cacheFolderName = '.cache';
 Constants.tempFolderName = 'temp';
 Constants.heftConfigurationFilename = 'heft.json';
