diff --git a/dist/unstable/httpapi/HttpApiClient.js b/dist/unstable/httpapi/HttpApiClient.js
index 0ef877b094106f5eaad8de39041d6df46bb0e59f..24583559dce5d694dd12919247d5f5dabc2e2fda 100644
--- a/dist/unstable/httpapi/HttpApiClient.js
+++ b/dist/unstable/httpapi/HttpApiClient.js
@@ -533,6 +533,20 @@ function getEncodePayloadSchemaFromBody(schema, method) {
   if (cached !== undefined) {
     return cached;
   }
+  if (HttpApiSchema.isStreamUint8Array(schema)) {
+    const out = $HttpBody.pipe(Schema.decodeTo(schema, SchemaTransformation.transformOrFail({
+      decode() {
+        return Effect.fail(new SchemaIssue.Forbidden({
+          message: "Encode only schema"
+        }));
+      },
+      encode(t) {
+        return Effect.succeed(HttpBody.stream(t, schema.contentType));
+      }
+    })));
+    bodyFromPayloadCache.set(ast, out);
+    return out;
+  }
   const encoding = HttpApiSchema.getPayloadEncoding(ast, method);
   const out = $HttpBody.pipe(Schema.decodeTo(schema, SchemaTransformation.transformOrFail({
     decode() {
diff --git a/dist/unstable/httpapi/HttpApiEndpoint.js b/dist/unstable/httpapi/HttpApiEndpoint.js
index dc8c1cb93fde0b2ed26d7da590a6930e8394f41d..189ce3189cf0ab96842abdf80f01b890e55791cc 100644
--- a/dist/unstable/httpapi/HttpApiEndpoint.js
+++ b/dist/unstable/httpapi/HttpApiEndpoint.js
@@ -341,6 +341,9 @@ function applyResponseEncoding(schema, encoding) {
   }
 }
 function transformPayload(schema, method) {
+  if (HttpApiSchema.isStreamSchema(schema)) {
+    return schema;
+  }
   const encoding = HttpApiSchema.getPayloadEncoding(schema.ast, method);
   switch (encoding._tag) {
     case "Json":
